File: app/src/main/java/com/itba/runningMate/mainpage/fragments/feed/FeedView.java
package com.itba.runningMate.mainpage.fragments.feed;

import com.itba.runningMate.domain.Run;

public interface FeedView {

    void goToPastRunsActivity();

    void setPastRunCardsNoText();

    void addRunToCard(int i, Run run);

    void disappearRuns(int abs);

    void disappearNoText();

    void setGoalTitle(int title);

    void setGoalSubtitle(int subtitle);

    void setGoalImage(int image);
}

File: app/src/main/java/com/itba/runningMate/mainpage/fragments/feed/FeedView.kt
package com.itba.runningMate.mainpage.fragments.feed

import com.itba.runningMate.domain.Achievements
import com.itba.runningMate.domain.Level
import com.itba.runningMate.domain.Run

interface FeedView {

    fun setPastRunCardsNoText()

    fun addRunToCard(i: Int, run: Run)

    fun disappearRuns(abs: Int)

    fun disappearNoText()

    fun showAchievements(achievements: List<Achievements>)

    fun showCurrentLevel(level: Level)

    fun launchLevelsActivity()

    fun launchAchievementsActivity()

    fun launchRunDetailActivity(runId: Long)

    fun launchPastRunsActivity()

    fun startLevelShimmerAnimation()

    fun stopLevelShimmerAnimation()

    fun startRecentActivityShimmerAnimation()

    fun stopRecentActivityShimmerAnimation()

}